vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -115,18 +115,24 @@ private void writeCacheHeaders(HttpServerRequest request, FileProps props) {
     if (cachingEnabled) {
       // We use cache-control and last-modified
       // We *do not use* etags and expires (since they do the same thing - redundant)
-      headers.set("cache-control", "public, max-age=" + maxAgeSeconds);
-      headers.set("last-modified", dateTimeFormatter.format(props.lastModifiedTime()));
+      writeNotExistingCacheHeader(headers, "cache-control", "public, max-age=" + maxAgeSeconds);
+      writeNotExistingCacheHeader(headers, "last-modified", dateTimeFormatter.format(props.lastModifiedTime()));
       // We send the vary header (for intermediate caches)
       // (assumes that most will turn on compression when using static handler)
       if (sendVaryHeader && request.headers().contains("accept-encoding")) {
-        headers.set("vary", "accept-encoding");
+        writeNotExistingCacheHeader(headers, "vary", "accept-encoding");
       }
     }
 
     // date header is mandatory
     headers.set("date", dateTimeFormatter.format(new Date()));
   }
+  
+  private void writeNotExistingCacheHeader(MultiMap headers, String key, String value) {
```

Comment:
We should have this method on the utilities class because we need this not just here but also for example on sockjs cors as it is affected by the same problem. If a header is set don't modify it.

Reply:
Done! Maybe, a method addIfAbsent could be added to MultiMap in future versions of core library, like the putIfAbsent method of Map interface. Thus, we wouldn't need the Utils method anymore.